### ISSUE TITLE
[Bug] Durations shown in the UI are not correct

### ISSUE THREAD

**Issue by nbro** [comet-ml/issue-tracking#192]
I created a public project: https://www.comet.ml/nbro/comparing-optimizers. I performed certain experiments. Each experiment uses the same code, but with different hyper-parameters. So I am creating an `Experiment` object inside a loop, for each iteration: you can see the source code here: https://github.com/nbro/comparative-study-between-optimizers/blob/master/main.py. 

If you now look at the entries in the "Duration" column of the Comet UI for this project, you will see increasing numbers: the first experiment which is shown (which is actually the last performed) lasted 2:00, the second lasted 4:03, etc. This is a bug either in the UI or in the recording of the durations of each experiment. It seems like the entries that are shown are the durations since the first experiment, because, if you look at the experiment at the bottom (which was actually the first experiment performed), it lasted 02:05:12, which is more or less the time it took to perform all experiments.

**Comment by Nimrod007** [comment]
Hi @nbro are all experiments running on `colab` ?

**Comment by nbro** [comment]
@Nimrod007 The mentioned experiments, yes. However, I also have this project https://www.comet.ml/nbro/comparing-optimizers, whose experiments were run locally on a Mac (not using any notebook, but just by calling Python directly).

**Comment by dsblank** [comment]
The comparing-optimizers code doesn't call experiment.end(), so that is why they may all have the same duration and same end time.

The optimizers-comparison looks like it might be a bug... thanks for reporting!

**Comment by nbro** [comment]
@dsblank In theory, AFAIK, there is no need to call `end()` explicitly, unless you are running the code in a notebook.

**Comment by dsblank** [comment]
If you are creating multiple experiments, and you want to have accurate duration and stop times, then you need to call experiment.end().

**Comment by dsblank** [comment]
I have tried to replicate but I cannot. I believe that your durations really are getting longer and longer. That can happen if you are increasing/decreasing batch_size, for example.

**Comment by nbro** [comment]
@dsblank No, this is definitely not the case. I change the batch size, but I don't necessarily increase it. Also, each experiment has only one (training) batch size, which may be different (or not) from batch size of the other experiments. You can have a look at it here: https://github.com/nbro/comparative-study-between-optimizers/blob/master/main.py#L38.

Also, I would like to note that the experiments more or less at most 1 minute. I am keeping track of the times manually. See this line: https://github.com/nbro/comparative-study-between-optimizers/blob/master/main.py#L105.

**Comment by dsblank** [comment]
Ok, that sounds like a deeper bug than I could find today. I have made a report about it and we will keep trying to replicate it and track it down. If you think of anything else that might be the cause let us know!

**Comment by nbro** [comment]
@dsblank Maybe I could help more if I knew how exactly you're keeping track of the durations. But if I discover something, I will let you know.

**Comment by dsblank** [comment]
Thanks for the notebook.... we'll try to replicate with that.

**Comment by dsblank** [comment]
No, we didn't use that... we came up with a minimal example running locally.

**Comment by tshrjn** [comment]
So, now is there a need to call `end` when running multiple experiments for example for hyperparam optimization?

**Comment by dsblank** [comment]
@tshrjn That might help make the ending times more accurate.
